### PR TITLE
[DOCS] Create and Publish an Application Integration Support Policy

### PR DESCRIPTION
**Pull request by kwcanuck**
In [DSB-184](https://greatexpectations.atlassian.net/browse/DSB-184), @joshzzheng requested the addition of a topic that clearly defines what application integrations are supported and who is responsible for resolving issues. This PR implements this request.

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] _Appropriate_ tests and docs have been updated

[DSB-184]: https://greatexpectations.atlassian.net/browse/DSB-184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+## Credential stores
+
+The following table defines the credential stores supported by GX and the community.
+
+| GX supported                             | Community supported                                                |
+|------------------------------------------|--------------------------------------------------------------------|
+| <i class="fas fa-check-circle" style={{color: "#28a745"}}></i> YAML<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> AWS Secrets Store<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Azure Blob Store<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> GCP Secrets Manager<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Azure Key Vault<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Databricks Secrets Store<br/><i class="fas fa-circle" style={{color: "#dc3545"}}></i> HashiCorp Vault    |               |
```
Comment (austiezr) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: I'm thinking HashiCorp shouldn't be under GX supported.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
...ass="fas fa-check-circle" style={{color: "#28a745"}}></i> Spark (maintenance mode)  | <i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Redshift<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> MSSQL<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> MySQL<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> SQLite<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Trino<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> AWS S3<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Google Cloud Storage<br/><i class="fas fa-check-circle" style={{color: "#28a745"}}></i> Azure Blog Storage<br/><i class="fas fa-circle" style={{color: "#ffc107"}}></i> Clickhouse<br/><i class="fas fa-circle" style={{color: "#ffc107"}}></i> Athena<br/><i class="fas fa-circle" style={{color: "#ffc107"}}></i> Dremio<br/><i class="fas fa-circle" style={{color: "#ffc107"}}></i> Teradata<br/><i class="fas fa-circle" style={{color: "#ffc107"}}></i> Vertica<br/><i class="fas fa-circle" style={{color: "#dc3545"}}></i> EMR Spark<br/><i class="fas fa-circle" style={{color: "#dc3545"}}></i> AWS Glue             |
```
Comment (austiezr) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: Should Databricks, Snowflake, and Bigquery be yellow here? Snowflake and BigQuery are certainly under partial test, and we support all three of them.

File: docs/docusaurus/docs/contributing/application_integration_support.md (hunk 1 above)
Comment (austiezr) [reply docs/docusaurus/docs/contributing/application_integration_support.md]: Also uncertain that all of these are under test -- certainly we don't have example scripts showing these under test for Databricks Secrets Store or Azure Blob Store.

**Review by austiezr — CHANGES_REQUESTED** [review]
Pinging @joshzzheng for verification on supported integrations here as well.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
+---
+title: Application support policy
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
title: Support policy
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+---
+
+A defined application support policy allows Great Expectations (GX) to better prioritize its response to application integration, operating system, and programming language issues. Formally defining the support policy also provides contributors with a better understanding of who is responsible for resolving application integration, operating system, and programming language issues.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
A defined support policy allows Great Expectations (GX) to prioritize its response to application integration, operating system, and programming language issues.

For production environments where uptime is critical, GX recommends using the applications, operating systems, and programming languages defined as GX-supported. Community-supported platforms such as Pandas, Spark, and SQLAlchemy allow you to integrate your existing datasets with GX Cloud and GX OSS.
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+The following are the types of support available for application integration, operating system, and programming language issues:
+
+- **GX supported** - The GX Developer Relations (DevRel) team triages, reproduces, and then assigns issues to the appropriate GX Engineering team. Issues with application integration, operating system, and programming languages in maintenance mode are reviewed, but new features or functionality are not implemented as part of the resolution.
+
+- **Community supported** - The GX DevRel team triages and reproduces issues, and then determines if GX should implement new features or functionality to resolve the issue. If the DevRel team determines that new features or functionality are not warranted, the GX community is responsible for issue resolution. If new features or functionality are required, DevRel assigns the issue to the appropriate GX Engineering team.
+
+## GX supported
+
+The following are the levels of support offered by GX for integrated applications, operating systems, and programming languages.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md:19]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
+---
+title: Application support policy
+---
+
+A defined application support policy allows Great Expectations (GX) to better prioritize its response to application integration, operating system, and programming language issues. Formally defining the support policy also provides contributors with a better understanding of who is responsible for resolving application integration, operating system, and programming language issues.
+
+## Support categories
+
+The following are the types of support available for application integration, operating system, and programming language issues:
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
The following are the levels of support provided by GX:

- GX-supported - integrations are tested throughout the development lifecycle, and actively maintained and updated when new versions of GX Cloud or GX OSS are released.

- Community-supported - integrations are successfully implemented by community members. They are not tested or updated by GX, and GX is not responsible for ensuring reliability or compatibility.
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+- **GX supported** - The GX Developer Relations (DevRel) team triages, reproduces, and then assigns issues to the appropriate GX Engineering team. Issues with application integration, operating system, and programming languages in maintenance mode are reviewed, but new features or functionality are not implemented as part of the resolution.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+- **Community supported** - The GX DevRel team triages and reproduces issues, and then determines if GX should implement new features or functionality to resolve the issue. If the DevRel team determines that new features or functionality are not warranted, the GX community is responsible for issue resolution. If new features or functionality are required, DevRel assigns the issue to the appropriate GX Engineering team.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+The following are the levels of support offered by GX for integrated applications, operating systems, and programming languages.
+
+### Operating systems
+
+The following table defines the operating systems supported by GX Cloud and GX OSS.
+
+| GX Cloud                     | GX OSS                        |
+|------------------------------|-------------------------------|
+| - Mac/Linux ¹                | - Mac/Linux                   |
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
 Mac/Linux ¹                | Mac/Linux                   | 
```

Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: I'd like to remove these bullets for single items, I think they make things harder to read,

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+| Data Source                              | - Snowflake          | - Snowflake<br/> - Generic SQL ¹               |
+| Configuration Stores                     | - N/A                | - Filesystem<br/> - AWS S3GCS Buckets<br/> - Azure Blob Store                |
+| Communication (notifications)            | - N/A ²               | - Slack (local) ³<br/> - email (local)<br/> - Microsoft Teams (local)<br/> - PagerDuty (local)                 |
+| Orchestrator                              | - Airflow ⁴           | - Airflow ⁴       |
+
+
+¹ Connection strings for various SQL Data Sources are supported, but dialect-specific SQL commands are not.<br/>
+² Support for Slack, email, and Zapier are planned for future releases.<br/>
+³ When used with the GX OSS library.<br/>
+⁴ Other Orchestrator applications that execute Python code are not supported.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
⁴ GX officially only supports Airflow, but it should work with any orchestrator that executes Python code.
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+### GX components
+
+The following table defines the GX components supported by GX Cloud and GX OSS.
+
+| Component                                | GX Cloud             | GX OSS          |
+|------------------------------------------|----------------------|-----------------|
+| Expectations                             | - See [Available Expectations](/docs/cloud/expectations/manage_expectations#available-expectations) ¹. | - See [Create Expectations](/docs/guides/expectations/expectations_lp) ¹.                |
+| GX Agent                                 | - All versions               | - N/A        |
+
+¹ The creation and execution of custom Expectations are supported, but inner queries are not.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+## Community supported
+
+The following integrated applications, operating systems, and programming languages are supported by the community.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md:78]:
```suggestion
```

I dont think we need this line. We already explained the two tiers above.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+### Operating systems
+
+The following table defines the operating systems supported by the community.
+
+| GX Cloud                       | GX OSS                        |
+|--------------------------------|-------------------------------|
+| - Mac/Linux ¹<br/>- Windows ²  | - Mac/Linux<br/>- Windows ²   |
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
| - Windows ²  | Windows ²   | 
```

Mac/Linux is GX-supported not community supported

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+The following table defines the operating systems supported by the community.
+
+| GX Cloud                       | GX OSS                        |
+|--------------------------------|-------------------------------|
+| - Mac/Linux ¹<br/>- Windows ²  | - Mac/Linux<br/>- Windows ²   | 
+
+¹ Required to run the GX Agent.<br/>
+² Untested and unsupported by GX
+
+### Python versions
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: We're not support previous versions at all. We should remove both Python versions and GX versions section.

File: docs/docusaurus/docs/contributing/application_integration_support.md (hunk 3 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/contributing/application_integration_support.md]: @joshzzheng - support policy for what? You need to specify the type of policy. This is a policy for application integrations, operating systems, and programming languages.

File: docs/docusaurus/docs/contributing/application_integration_support.md (hunk 5 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/contributing/application_integration_support.md:19]: @joshzzheng, we can't use empty headings. See heading and title format in the Google developer documentation style guide (https://developers.google.com/style/headings#heading-and-title-format). We need an introductory statement after the heading.

File: docs/docusaurus/docs/contributing/application_integration_support.md (hunk 11 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/contributing/application_integration_support.md:76]: @joshzzheng, we can't use empty headings. See heading and title format in the Google developer documentation style guide (https://developers.google.com/style/headings#heading-and-title-format). We need an introductory statement after the heading.

File: docs/docusaurus/docs/contributing/application_integration_support.md (hunk 12 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/contributing/application_integration_support.md:78]: @joshzzheng, we can't use empty headings. See heading and title format in the Google developer documentation style guide (https://developers.google.com/style/headings#heading-and-title-format). We need an introductory statement after the heading.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
+---
+title: Integration support policy
+---
+
+A defined integration support policy allows Great Expectations (GX) to prioritize its response to application integration, operating system, and programming language issues.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+
+For production environments where uptime is critical, GX recommends using the applications, operating systems, and programming languages defined as GX-supported. Community-supported platforms such as Pandas, Spark, and SQLAlchemy allow you to integrate your existing datasets with GX Cloud and GX OSS.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
For production environments, GX recommends using the integrations defined as GX-supported. 

GX uses libraries such as Pandas, Spark, and SQLAlchemy to integrate with a wide range of data sources. This also allows you to deploy GX with Community-supported integrations.
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+| Application Type                         | GX Cloud             | GX OSS          |
+|------------------------------------------|----------------------|-----------------|
+| Data Source                              | Snowflake          | - Redshift<br/>- MSSQL<br/>- MySQL<br/>- SQLite<br/>- Trino<br/>- AWS S3<br/>- Google Cloud Storage<br/>- Azure Blog Storage<br/>- Clickhouse ¹<br/>- Athena ¹<br/>- Dremio ¹<br/>- Teradata ¹<br/>- Vertica ¹<br/>- EMR Spark ²<br/>- AWS Glue ²               |
+| Configuration Stores                     | N/A                | Azure Blob Store                |
+| Communication (notifications)            | N/A                | - Slack (local) ³<br/> - email (local)<br/> - Microsoft Teams (local)<br/> - PagerDuty (local)<br/>- Opsgenie<br/>- Amazon SNS<br/>- General API<br/>- DataHub ⁴                   |
+| Orchestrator                              | Airflow ⁵           | - Airflow<br/>- Prefect<br/>- Dagster ⁶<br/>- Flyte ⁶<br/>- mage.ai ⁶<br/>- Github Action ⁶        |
+
+
+¹ Partially tested and supported by GX.<br/>
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+² Untested and unsupported by GX.<br/>
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: delete

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+| Windows ¹                      | Windows ¹                     | 
+
+¹ Untested and unsupported by GX.
+
+### Integrations
+
+The following table defines the GX Cloud and GX OSS integrations supported by the community.
+
+
+| Application Type                         | GX Cloud             | GX OSS          |
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: All GX Cloud is N/A.

Explain why N/A. All Community-supported integrations are applicable only to GX OSS.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+| N/A                         | - 0.17<br/> - 0.18              | 
+
+### Integrations
+
+The following table defines the supported GX Cloud and GX OSS integrations.
+
+
+| Application Type                         | GX Cloud             | GX OSS          |
+|------------------------------------------|----------------------|-----------------|
+| Data Source                              | Snowflake           | - Snowflake<br/> - Generic SQL ¹               |
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: Support for Bigquery coming soon.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+
+For production environments, GX recommends using the integrations defined as GX-supported. 
+
+GX uses libraries such as Pandas, Spark, and SQLAlchemy to integrate with a wide range of data sources. This also allows you to deploy GX with Community-supported integrations.
+
+## Support categories
+
+The following are the levels of support provided by GX:
+
+- GX-supported - integrations are tested throughout the development lifecycle, and actively maintained and updated when new versions of GX Cloud or GX OSS are released.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
- GX-supported - integrations are tested throughout the development lifecycle, and are actively maintained and updated when new versions of GX Cloud or GX OSS are released.
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+
+- Community-supported - integrations are implemented by community members. They are not tested or updated by GX, and GX is not responsible for ensuring reliability or compatibility.
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]: These integrations are are not just implemented by community members. Some of these were created by GX. We also test and update some of them. We've also seen successful implementations of some of these. I'm ok with the last piece around responsible for reliability or compatibility.

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+### Integrations
+
+The following table lists the GX Cloud and GX OSS integrations supported by the community.
+
+
+| Integration Type                         | GX Cloud             | GX OSS          |
+|------------------------------------------|----------------------|-----------------|
+| Data Source                              | N/A          | Pandas<br/>Spark<br/>Databricks (Spark)<br/>Databricks (SQL)<br/>Trino<br/>Clickhouse<br/>Dremio<br/> Teradata<br/>Vertica<br/>EMR Spark<br/>AWS Glue<br/>Google Cloud Storage<br/>Azure Blog Storage<br/>AWS S3|
+| Notifications                             | N/A            | Opsgenie<br/>Amazon SNS<br/>DataHub |
+| Orchestrator                              | N/A            | Prefect<br/>Dagster <br/>Flyte <br/>mage.ai  |
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
| Orchestrators                             | N/A            | Prefect<br/>Dagster <br/>Flyte <br/>mage.ai  |
```

File: docs/docusaurus/docs/contributing/application_integration_support.md [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+¹ Untested and unsupported by GX.
+
+### Integrations
+
+The following table lists the GX Cloud and GX OSS integrations supported by the community.
+
+
+| Integration Type                         | GX Cloud             | GX OSS          |
+|------------------------------------------|----------------------|-----------------|
+| Data Source                              | N/A          | Pandas<br/>Spark<br/>Databricks (Spark)<br/>Databricks (SQL)<br/>Trino<br/>Clickhouse<br/>Dremio<br/> Teradata<br/>Vertica<br/>EMR Spark<br/>AWS Glue<br/>Google Cloud Storage<br/>Azure Blog Storage<br/>AWS S3|
```
Comment (joshzzheng) [inline docs/docusaurus/docs/contributing/application_integration_support.md]:
```suggestion
| Data Sources                              | N/A          | Pandas<br/>Spark<br/>Databricks (Spark)<br/>Databricks (SQL)<br/>Trino<br/>Clickhouse<br/>Dremio<br/> Teradata<br/>Vertica<br/>EMR Spark<br/>AWS Glue<br/>Google Cloud Storage<br/>Azure Blog Storage<br/>AWS S3|
```

**Review by joshzzheng — APPROVED** [review]

**Review by austiezr — APPROVED** [review]
🚀

**Pull request merged**